### ISSUE TITLE
Swarm-kit containers stuck in initializing state when a 5 node cluster is started.

### ISSUE THREAD

**Issue by soumyalj** [rancher/rancher#6145]
Rancher Version: v1.2.0-pre3

Docker Version:1.12.1

OS and where are the hosts located? (cloud, bare metal, etc):Ubuntu 14.0.4, Cloud

Steps to Reproduce:
1. Create a 5 host swarm cluster by using the catalog https://github.com/LLParse/swarmkit-catalog. Use Digital Ocean hosts.
Set CATTLE_AEGNT_IP=internal ip for the hosts. Once the catalog is launched, the swarm cluster does not come up successfully. The swarm kit containers are in initializing state.

Logs for the swarmkit-mon container

```
root@soumyadoswarm1:~# docker logs -f cc9
+ META_URL=http://rancher-metadata.rancher.internal/2015-12-19
+ SERVICE_NAME=swarmkit-mon
+ MANAGER_SCALE=5
++ curl -s http://rancher-metadata.rancher.internal/2015-12-19/self/host/agent_ip
+ AGENT_IP=10.134.2.198
+ '[' 10.134.2.198 == '' ']'
++ curl -s http://rancher-metadata.rancher.internal/2015-12-19/services/swarmkit-mon/uuid
+ SERVICE_UUID=b1230b74-d4e8-4cce-b8ed-4d9cc3e0983d
++ curl -s http://rancher-metadata.rancher.internal/2015-12-19/self/host/uuid
+ HOST_UUID=40670a82-a142-44b6-91b6-193611141c77
+ true
+ node
++ get_leader
++ local lowest_index lowest_ip
+ giddyup health -p 2378 --check-command /opt/rancher/health.sh
+++ containers
++++ curl -s http://rancher-metadata.rancher.internal/2015-12-19/services/swarmkit-mon/containers
time="2016-10-04T23:40:41Z" level=info msg="Listening on port: 2378" 
+++ echo 0=swarmkit_swarmkit-mon_1 1=swarmkit_swarmkit-mon_2 2=swarmkit_swarmkit-mon_3 3=swarmkit_swarmkit-mon_4 4=swarmkit_swarmkit-mon_5
++ for container in '$(containers)'
+++ cut -d= -f2
+++ echo 0=swarmkit_swarmkit-mon_1
++ c=swarmkit_swarmkit-mon_1
+++ container_create_idx swarmkit_swarmkit-mon_1
++++ curl -s http://rancher-metadata.rancher.internal/2015-12-19/services/swarmkit-mon/containers/swarmkit_swarmkit-mon_1/create_index
+++ echo 1
++ create_index=1
++ '[' '' == '' ']'
++ lowest_index=1
+++ container_ip swarmkit_swarmkit-mon_1
++++ curl -s http://rancher-metadata.rancher.internal/2015-12-19/services/swarmkit-mon/containers/swarmkit_swarmkit-mon_1/primary_ip
+++ echo 10.134.2.198
++ lowest_ip=10.134.2.198
++ for container in '$(containers)'
+++ cut -d= -f2
+++ echo 1=swarmkit_swarmkit-mon_2
++ c=swarmkit_swarmkit-mon_2
+++ container_create_idx swarmkit_swarmkit-mon_2
++++ curl -s http://rancher-metadata.rancher.internal/2015-12-19/services/swarmkit-mon/containers/swarmkit_swarmkit-mon_2/create_index
+++ echo 2
++ create_index=2
++ '[' 1 == '' ']'
++ '[' 2 -lt 1 ']'
++ for container in '$(containers)'
+++ cut -d= -f2
+++ echo 2=swarmkit_swarmkit-mon_3
++ c=swarmkit_swarmkit-mon_3
+++ container_create_idx swarmkit_swarmkit-mon_3
++++ curl -s http://rancher-metadata.rancher.internal/2015-12-19/services/swarmkit-mon/containers/swarmkit_swarmkit-mon_3/create_index
+++ echo 3
++ create_index=3
++ '[' 1 == '' ']'
++ '[' 3 -lt 1 ']'
++ for container in '$(containers)'
+++ cut -d= -f2
+++ echo 3=swarmkit_swarmkit-mon_4
++ c=swarmkit_swarmkit-mon_4
+++ container_create_idx swarmkit_swarmkit-mon_4
++++ curl -s http://rancher-metadata.rancher.internal/2015-12-19/services/swarmkit-mon/containers/swarmkit_swarmkit-mon_4/create_index
+++ echo 4
++ create_index=4
++ '[' 1 == '' ']'
++ '[' 4 -lt 1 ']'
++ for container in '$(containers)'
+++ cut -d= -f2
+++ echo 4=swarmkit_swarmkit-mon_5
++ c=swarmkit_swarmkit-mon_5
+++ container_create_idx swarmkit_swarmkit-mon_5
++++ curl -s http://rancher-metadata.rancher.internal/2015-12-19/services/swarmkit-mon/containers/swarmkit_swarmkit-mon_5/create_index
+++ echo 5
++ create_index=5
++ '[' 1 == '' ']'
++ '[' 5 -lt 1 ']'
++ echo 10.134.2.198
+ '[' 10.134.2.198 == 10.134.2.198 ']'
++ local_node_state
+++ jq -r .Swarm.LocalNodeState
+++ curl -s --unix-socket /var/run/docker.sock http::/info
++ echo inactive
+ '[' inactive == inactive ']'
+ bootstrap_node
+ '[' '' == '' ']'
+ docker swarm init
Error response from daemon: could not choose an IP address to advertise since this system has multiple addresses on interface eth0 (104.236.137.117 and 10.12.0.5) - specify one with --advertise-addr
+ publish_tokens
```

**Comment by LLParse** [comment]
Fixed in template version `1.12-rancher2`

**Comment by soumyalj** [comment]
Tested with template version  1.12-rancher2. 5 host swarm cluster came up successfully.
